internal/mode/static/manager.go

```diff
@@ -158,6 +161,7 @@ func StartManager(cfg config.Config) error {
 
 	var ngxPlusClient *ngxclient.NginxClient
```

Comment:
@miledxz can you change this type to nginxPlusClient interface defined in the nginx runtime package?

The functional tests fail because the function IsPlus on the nginx runtime manager returns true even when the nginxPlusClient is nil. This is due to a subtlety around interface types.

If you pass a nil `*ngxclient.NginxClient` as the type `nginxPlusClient`, it will not evaluate as nil. Take a look at this go playground example: https://goplay.tools/snippet/sDiRga5qSqm

We need to change this type to the interface nginxPlusClient type to fix this. This means you will need to export the nginxPlusClient interface and modify the collectors.NewNginxPlusMetricsCollector function to accept the interface instead of the actual client. 

Apologies for not catching this sooner.

Reply:
@miledxz there's a couple errors in the pipeline that need to be fixed:

```
cannot find package with target: nginxPlusClient
Writing `FakeNginxPlusClient` to `runtimefakes/fake_nginx_plus_client.go`... exit status 1
```

```
  Error: internal/mode/static/metrics/collectors/nginx.go:33:3: cannot use plusClient (variable of type "github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/nginx/runtime".NginxPlusClient) as *"github.com/nginxinc/nginx-plus-go-client/client".NginxClient value in argument to nginxCollector.NewNginxPlusCollector: need type assertion (typecheck)
```